packages/wrangler/src/versions/deploy.ts

```diff
@@ -205,6 +209,61 @@ function getConfig(
 	return config;
 }
 
+/**
+ * Prompts the user for confirmation when overwriting the latest deployment, given that it's split.
+ */
+export async function confirmLatestDeploymentOverwrite(
+	accountId: string,
+	scriptName: string
+) {
+	try {
+		const latest = await fetchLatestDeployment(accountId, scriptName);
+		if (latest && latest.versions.length >= 2) {
+			// Fetch the version details.
+
+			const versionCache: VersionCache = new Map();
+			const versionIds = latest.versions.map((v) => v.version_id);
+			await fetchVersions(accountId, scriptName, versionCache, ...versionIds);
+
+			const versions = latest.versions.map((v) => {
+				const version = versionCache.get(v.version_id);
+				assert(version);
+				return version;
+			});
+
+			const traffic = new Map(
+				latest.versions.map((v) => [v.version_id, v.percentage])
+			);
```

Comment:
What I didn't like about `printLatestDeployment` is that it would fetch the deployment again when I already had it. Ended up refactoring to a function named `printDeployment` that lets you pass and print any deployment as a parameter.

Reply:
I had an idea to pass a callback param like `skipIf(deployment: ApiDeployment) => boolean` which would enable flow control using the one api request

but this is fine for now, and I can refactor in a follow up if we decide it's worth it